addons/account_payment/models/payment_transaction.py

```diff
@@ -161,17 +161,40 @@ def _create_payment(self, **extra_create_values):
             'ref': reference,
             **extra_create_values,
         }
-        payment = self.env['account.payment'].create(payment_values)
-        payment.action_post()
-
-        # Track the payment to make a one2one.
-        self.payment_id = payment
 
         # Reconcile the payment with the source transaction's invoices in case of a partial capture.
         if self.operation == self.source_transaction_id.operation:
             invoices = self.source_transaction_id.invoice_ids
         else:
             invoices = self.invoice_ids
+
+        today = fields.Date.context_today(self)
+        invoices_with_epd = invoices.filtered(lambda inv: inv._is_eligible_for_early_payment_discount(inv.currency_id, today))
+        if invoices_with_epd:
+            wizard = self.env['account.payment.register'].with_context(
+                active_model='account.move',
+                active_ids=invoices_with_epd.ids,
+            ).create({
+                'payment_date': today,
+                'early_payment_discount_mode': True,
+            })
+            batch_result = wizard._get_batches()[0]
+            write_off_line_vals = wizard._get_write_off_vals_from_wizard(batch_result)
```

Comment:
There is no point to use the wizard if it's only to call an helper that is not used by the wizard itself.
I don't think you can have multiple invoices. Otherwise, there will be plenty ways to break the flow since you can't ensure you can pay them only with a single payment.
Instead, why not just call the payment register wizard all the time?
```py
wizard = self.env['account.payment.register'].with_context(
    active_model='account.move',
    active_ids=invoices.ids,
).create({
    'journal_id': self.provider_id.journal_id.id,
    'payment_date': fields.Date.context_today(self),
    'amount': abs(self.amount),
    'communication': reference,
    'partner_type': 'customer',
    'payment_method_line_id': payment_method_line.id,
    'partner_id': self.partner_id.commercial_partner_id.id,
    'group_payment': True,
})
batch_result = wizard._get_batches()[0]
payment_values = wizard._create_payment_vals_from_batch(batch_result)
self.payment_id = payment = self.env['account.payment'].create({
    **payment_values,
    'payment_token_id': self.token_id.id,
    'payment_transaction_id': self.id,
    **extra_create_values,
})
payment.action_post()
```